python/lsst/cp/pipe/cpFilterScan.py

```diff
@@ -0,0 +1,377 @@
+        self.makeSubtask("combine")
+
+    def run(self, inputExpHandles, inputElectrometerHandles=None):
+        """Construct filter scan from the header and visit info of processed
+        exposures.
+
+        Parameters
+        ----------
+        inputExpHandles : `list` [`lsst.daf.butler.DeferredDatasetHandle`]
+            Input list of exposure handles to combine.
+        inputElectrometerHandles : `list` [`lsst.daf.butler.DeferredDatasetHandle`], optional # noqa W505
+            Input list of electrometer/photodiode measurement handles
+            to combine.
+
+        Returns
+        -------
+        results : `lsst.pipe.base.Struct`
+            The results struct containing:
+
+            ``outputData``
+                Final combined filter scan, with a single table
+                containing the measured throughput for all input
+                filters at the various wavelength values indicated in
+                the exposure's observationReason
+                (`astropy.table.Table`).
+        """
+        filterScanResults = {}
+        filterSet = set()
+        efdClient = CpEfdClient(efdInstance=self.config.efdClientInstance)
+        monochromatorData = efdClient.getEfdMonochromatorData()
+
+        for handle in inputExpHandles:
+            visitInfo = handle.get(component="visitInfo")
+            metadata = handle.get(component="metadata")
+
+            physical_filter = handle.dataId['physical_filter']
+            exposureId = handle.dataId['exposure']
+            exposureTime = visitInfo.exposureTime
+
+            values = [v for k, v in metadata.items() if "LSST ISR FINAL MEAN" in k]
+
+            scaleFactor = exposureTime
+            if self.config.useElectrometer:
+                for eHandle in inputElectrometerHandles:
+                    if eHandle.dataId.exposure == exposureId:
+                        charge = eHandle.get().integrate()
+                        if charge > 0.0 and np.isfinite(charge):
+                            scaleFactor /= charge
+                        break
+
+            scan = {
+                'physical_filter': physical_filter,
+                'scale': scaleFactor,
+                'flux': np.median(values),
+            }
+            filterSet.add(physical_filter)
+
+            _, key = efdClient.parseMonochromatorStatus(monochromatorData,
+                                                        visitInfo.date.toString(DateTime.TAI))
+            key = float(key)
+            if key in filterScanResults:
+                filterScanResults[key].append(scan)
+            else:
+                filterScanResults[key] = [scan]
+
+        filterScan = []
+        for wavelength in filterScanResults.keys():
+            scans = filterScanResults[wavelength]
+
+            referenceScan = [x for x in scans if x['physical_filter'] == self.config.referenceFilter]
+            if len(referenceScan) == 0:
+                # No reference scan at this wavelength.
```

Comment:
This should probably log a warning, since the plan would be to have pairs, correct?

Reply:
Definitely, this is an oversight.  I've also updated the other warning with the wavelength.